homeassistant/loader.py

```diff
@@ -734,7 +741,12 @@ def integration_type(
     @cached_property
     def import_executor(self) -> bool:
         """Import integration in the executor."""
-        return self.manifest.get("import_executor") or False
+        # If the integration does not explicitly set import_executor, we default to
+        # True if it's a built-in integration and False if it's a custom integration.
+        # In the future, we want to default to True for all integrations.
+        return self.manifest.get(
+            "import_executor", self.pkg_path.startswith("homeassistant.components")
```

Comment:
Can we always default to True for when a user is running dev? This will help custom integration developers during development.

Reply:
good idea

Reply:
I change my mind on how to roll this out for CCs as I can't come up with a good justification for continuing to have a default of blocking the loop since they already have an escape hatch.  

https://github.com/home-assistant/core/pull/112127#issuecomment-1975744378